projector-plugin/src/main/kotlin/ProjectorService.kt

```diff
@@ -38,11 +42,39 @@ enum class EnabledState {
   HAS_VM_OPTIONS_AND_ENABLED,
 }
 
-object ProjectorService {
+class ProjectorConfig : PersistentStateComponent<ProjectorConfig> {
+  var selectedHost: String? = null
+  var selectedPort: String? = null
 
+  override fun getState(): ProjectorConfig? {
+    return this
+  }
+
+  override fun loadState(state: ProjectorConfig) {
+    selectedHost = state.selectedHost
+    selectedPort = state.selectedPort
+  }
+}
+
+@State(name = "Projector", storages = [Storage("ProjectorConfig.xml")])
+class ProjectorService : PersistentStateComponent<ProjectorConfig> {
+  private var config: ProjectorConfig = ProjectorConfig()
   private val logger = Logger.getInstance(ProjectorService::class.java)
   private val plugin = PluginManager.getPlugin(PluginId.getId("org.jetbrains.projector-plugin"))!!
 
+  val selectedHost: String?
+    get() = config.selectedHost
+
+  val selectedPort: String?
+    get() = config.selectedPort
+
+  var currentSession: Session? = null
+    set(value) {
+      field = value
+      config.selectedHost = value!!.host
```

Comment:
Super inconsistent. You set null session to the `currentSession` property but require it to be not null when set host and port while they both are nullable and can be reset without any problems (config.host = value?.host)